iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/IcebergMajorQueryCompactor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive.compaction;
+
+import java.io.IOException;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.DriverUtils;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.hive.ql.txn.compactor.CompactorContext;
+import org.apache.hadoop.hive.ql.txn.compactor.QueryCompactor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergMajorQueryCompactor extends QueryCompactor  {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergMajorQueryCompactor.class.getName());
+
+  @Override
+  public boolean run(CompactorContext context) throws IOException, HiveException, InterruptedException {
+
+    String compactTableName = context.getTable().getTableName();
+    LOG.debug("Executing compaction query for table {}", compactTableName);
+
+    String insertIntoTempIcebergTable = String.format("insert into %s select * from %s",
```

Comment:
Why not `insret overwrite`？

Reply:
IOW causes problems if an Iceberg table had partition evolution and it is not supported ([HIVE-26133](https://issues.apache.org/jira/browse/HIVE-26133)). The workaround in such case is to do truncate before IOW.
And this is what is done in this PR.
In case of compaction request, the commitWrite method was extended to do these 2 operations in a single transaction: 

1. Start a new transaction.
2. Delete all existing files in the manifest (equivalent to truncate table operation).
3. Add new files to the manifest.
4. Commit transaction.

Reply:
please use IOW, but handle it properly in commitOverwrite